File: dev-docs/snippets/workflows/list-step-runs.ts
import type { JsBaoClient } from "js-bao-wss-client";

// List the per-step records for one run (debugging: input/output/error/timing).
export async function listStepRuns(client: JsBaoClient, runId: string) {
  // #region example
  const steps = await client.workflows.listStepRuns({ runId });
  // steps.items: WorkflowStepRunRecord[]
  // #endregion example
  return steps;
}

File: dev-docs/snippets/workflows/list-step-runs.swift
import JsBaoClient

// List the per-step records for one run (debugging: input/output/error/timing).
func listStepRuns(client: JsBaoClient, runId: String) async throws {
  // #region example
  let steps = try await client.workflows.listStepRuns(runId: runId)
  // steps.items: [WorkflowStepRunRecord]
  // #endregion example
  _ = steps
}
